# Actividad Practica 1

La presente actividad tiene como objetivo la obtención de 3 datasets potenciales para realizar el proyecto final del curso de Data Science I de Coderhouse. Las características requeridas de cada dataset son:

* 15 columnas mínimo
* 2000 filas mínimo

Los datasets seleccionados que se describirán en las secciones siguientes son:

1. **household_income**: Dataset que muestra la información de varias variables que pueden afectar el resultado de la renta anual de una familia. Puede encontrar el dataset en el siguiente [Link](https://www.kaggle.com/datasets/stealthtechnologies/regression-dataset-for-household-income-analysis).
2. **spotify_songs_2024**: Dataset que muestra la información del desempeño de canciones alojadas en spotify y tambien en otras plataformas, en términos de escuchas, vistas, playlists, menciones, etc. Puede encontrar el dataset en el siguiente [Link](https://www.kaggle.com/datasets/nelgiriyewithana/most-streamed-spotify-songs-2024).
3. **athlete_events.csv**: Dataset que muestra la información de resultados individuales de todos los atletas olimpicos desde el inicio de los juegos olímpicos modernos, tanto de verano como de invierno. Puede encontrar el dataset en el siguiente [Link](https://www.kaggle.com/datasets/heesoo37/120-years-of-olympic-history-athletes-and-results).

## Importación de librerias

In [27]:
import pandas as pd

# Dataset #1: *household_income*

## Descripción del dataset

Corresponde a un dataset con varios factores demográficos y socioeconomicos que influencian el ingreso familiar anual. Las columnas que componen el dataset se describen a continuación.

* Age: Edad del jefe de hogar (18 to 70 years).

* Education Level: Nivel de educación más alto (High School, Bachelor's, Master's, Doctorate).

* Occupation: Tipo de ocupación (Healthcare, Education, Technology, Finance, Others).

* Number of Dependents: Numero de personas dependientes del jefe de hogar (0 to 5).

* Location: Tipo de zona residencial (Urban, Suburban, Rural).

* Work Experience: Años de experiencia laboral (0 to 50 years).

* Marital Status: Estado civil del jefe de hogar (Single, Married, Divorced).

* Employment Status: Situacion laboral del jefe de hogar (Full-time, Part-time, Self-employed).

* Household Size: Numero total de miembros del hogar (1 to 7).

* Homeownership Status: Situación de posesión de la vivienda (Own, Rent).

* Type of Housing: Tipo de vivienda (Apartment, Single-family home, Townhouse).

* Gender: Género del jefe de hogar (Male, Female).

* Primary Mode of Transportation: Modo de transporte del jefe de hogar (Car, Public transit, Biking, Walking).

* Annual Household Income: Ingreso anual del hogar en USD

## Importación del dataset

In [28]:
df = pd.read_csv("datasets/household_income.csv")
df.head()

Age Education_Level  Occupation  Number_of_Dependents Location  \
0   56        Master's  Technology                     5    Urban   
1   69     High School     Finance                     0    Urban   
2   46      Bachelor's  Technology                     1    Urban   
3   32     High School      Others                     2    Urban   
4   60      Bachelor's     Finance                     3    Urban   

   Work_Experience Marital_Status Employment_Status  Household_Size  \
0               21        Married         Full-time               7   
1                4         Single         Full-time               7   
2                1         Single         Full-time               7   
3               32        Married         Full-time               1   
4               15        Married     Self-employed               4   

  Homeownership_Status     Type_of_Housing  Gender  \
0                  Own           Apartment    Male   
1                  Own           Apartment    Male   
2                  Own  Single-family home  Female   
3                  Own           Apartment  Female   
4                  Own           Townhouse    Male   

  Primary_Mode_of_Transportation  Income  
0                 Public transit   72510  
1                         Biking   75462  
2                            Car   71748  
3                            Car   74520  
4                        Walking  640210

In [29]:
df.shape

(10000, 14)

## Descripción de las variables principales

### Información y tipos de datos

El dataset se compone de 14 columnas y 10000 entradas no nulas. Éste dataset ya se encuentra limpio y los tipos de datos corresponden a la información proporcionada en cada columna.

In [30]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10000 entries, 0 to 9999
Data columns (total 14 columns):
 #   Column                          Non-Null Count  Dtype 
---  ------                          --------------  ----- 
 0   Age                             10000 non-null  int64 
 1   Education_Level                 10000 non-null  object
 2   Occupation                      10000 non-null  object
 3   Number_of_Dependents            10000 non-null  int64 
 4   Location                        10000 non-null  object
 5   Work_Experience                 10000 non-null  int64 
 6   Marital_Status                  10000 non-null  object
 7   Employment_Status               10000 non-null  object
 8   Household_Size                  10000 non-null  int64 
 9   Homeownership_Status            10000 non-null  object
 10  Type_of_Housing                 10000 non-null  object
 11  Gender                          10000 non-null  object
 12  Primary_Mode_of_Transportation  10000 non-null 

### Descripción estadistica del dataset

In [31]:
# Descripción de los datos numericos
df.describe()

Age  Number_of_Dependents  Work_Experience  Household_Size  \
count  10000.000000          10000.000000     10000.000000    10000.000000   
mean      44.021700              2.527000        24.858800        3.989600   
std       15.203998              1.713991        14.652622        2.010496   
min       18.000000              0.000000         0.000000        1.000000   
25%       31.000000              1.000000        12.000000        2.000000   
50%       44.000000              3.000000        25.000000        4.000000   
75%       57.000000              4.000000        37.000000        6.000000   
max       70.000000              5.000000        50.000000        7.000000   

             Income  
count  1.000000e+04  
mean   8.168382e+05  
std    1.821089e+06  
min    3.104400e+04  
25%    6.844600e+04  
50%    7.294300e+04  
75%    3.506675e+05  
max    9.992571e+06

In [32]:
df["Income"].median()

72943.0

In [33]:
# Descripción de los datos no-numéricos
df.describe(exclude="number")

Education_Level  Occupation Location Marital_Status Employment_Status  \
count            10000       10000    10000          10000             10000   
unique               4           5        3              3                 3   
top         Bachelor's  Healthcare    Urban        Married         Full-time   
freq              4058        3035     7037           5136              5004   

       Homeownership_Status     Type_of_Housing Gender  \
count                 10000               10000  10000   
unique                    2                   3      2   
top                     Own  Single-family home   Male   
freq                   6018                4055   5123   

       Primary_Mode_of_Transportation  
count                           10000  
unique                              4  
top                    Public transit  
freq                             4047

### Observaciones sobre la información contenida en el dataset

Se pueden recoger las siguientes observaciones del dataset, en esta etapa preliminar.

- Existen 13 variables que se pueden relacionar con el ingreso, siendo 4 numéricas y 9 categóricas
- La edad de los encuestados se encuentra entre los 18 y los 70 años, con un promedio de **44.02**
- La cantidad de dependientes se encuentra entre 0 y 5 personas con un promedio de **2.52**
- La experiencia laboral se encuentra entre 0 y 50 años, con un promedio de **24.85**
- La cantidad total de integrantes se encuentra entre 1 y 7 personas, con un promedio de **3.98**
- El ingreso anual se encuentra entrelos 31044 USD y los 9992571 USD anuales, con un promedio de **816838 USD**. La mediana de esta variables es **72943 USD**.

# Dataset #2: *spotify_songs_2024*

## Descripción del dataset

Corresponde a un dataset con varios datos sobre el rendimiento de canciones de spotify en diversas plataformas, incluyendo spotify, youtube, tiktok, etc. Las columnas que componen el dataset se describen a continuación.

* Track Name: Nombre de la canción.
* Album Name: Nombre del album.
* Artist: Nombre del artista.
* Release Date: Fecha de lanzamiento.
* ISRC: Codigo de grabación estandar internacional (International Standard Recording Code).
* All Time Rank: Ranking de la canción basada en su popularidad global.
* Track Score: Puntuación asignada a la cancion basada en varios factores.
* Spotify Streams: Numero total de streans en spotify.
* Spotify Playlist Count: Numero de playlist que incluyen a la cancion.
* Spotify Playlist Reach: Alcance de la cancion a traves de las playlists de spotify.
* Spotify Popularity: Puntaje de popularidad de la cancion en spotify.
* YouTube Views: Numero total de vistas del video oficial de la cancion en youtube.
* YouTube Likes: Numero total de likes del video oficial de la cancion en youtube.
* TikTok Posts: Numero total de posteos de TikTok que contienen la cancion.
* TikTok Likes: Numero total de likes de los posteos de TikTok que contienen la cancion.
* TikTok Views: Numero total de vistas de los posteos de TikTok que contienen la cancion.
* YouTube Playlist Reach: Alcance de la cancion a traves de las playlists de YouTube.
* Apple Music Playlist Count: Numero de playlists de Apple Music que contienen la cancion.
* AirPlay Spins: Numero de veces que la cancion ha sido reproducida en estaciones de radio.
* SiriusXM Spins: Numero de veces que la cancion ha sido reproducida en SiriusXM.
* Deezer Playlist Count: Numero de playlists de Deezer que contienen la cancion.
* Deezer Playlist Reach: Alcance de la cancion a traves de las playlists de Deezer.
* Amazon Playlist Count: Numero de playlists de Amazon Music que contienen la cancion.
* Pandora Streams: Numero total de transmisiones en Pandora.
* Pandora Track Stations: Numero total de estaciones de Pandora que han transmitido la cancion.
* Soundcloud Streams: Numero total de transmisiones en Soundcloud.
* Shazam Counts: Numero total de veces que la cancion se ha Shazameado.
* TIDAL Popularity: Puntaje de popularidad de la cancion en TIDAL.
* Explicit Track: Indica si la cancion contiene lenguaje explicito.

## Importación del dataset

In [34]:
df = pd.read_csv("datasets/spotify_songs_2024.csv",encoding="unicode_escape")
df.head()

Track                    Album Name          Artist  \
0         MILLION DOLLAR BABY  Million Dollar Baby - Single   Tommy Richman   
1                 Not Like Us                   Not Like Us  Kendrick Lamar   
2  i like the way you kiss me    I like the way you kiss me         Artemas   
3                     Flowers              Flowers - Single     Miley Cyrus   
4                     Houdini                       Houdini          Eminem   

  Release Date          ISRC All Time Rank  Track Score Spotify Streams  \
0    4/26/2024  QM24S2402528             1        725.4     390,470,936   
1     5/4/2024  USUG12400910             2        545.9     323,703,884   
2    3/19/2024  QZJ842400387             3        538.4     601,309,283   
3    1/12/2023  USSM12209777             4        444.9   2,031,280,633   
4    5/31/2024  USUG12403398             5        423.3     107,034,922   

  Spotify Playlist Count Spotify Playlist Reach  ...  SiriusXM Spins  \
0                 30,716            196,631,588  ...             684   
1                 28,113            174,597,137  ...               3   
2                 54,331            211,607,669  ...             536   
3                269,802            136,569,078  ...           2,182   
4                  7,223            151,469,874  ...               1   

  Deezer Playlist Count Deezer Playlist Reach Amazon Playlist Count  \
0                  62.0            17,598,718                 114.0   
1                  67.0            10,422,430                 111.0   
2                 136.0            36,321,847                 172.0   
3                 264.0            24,684,248                 210.0   
4                  82.0            17,660,624                 105.0   

  Pandora Streams Pandora Track Stations Soundcloud Streams  Shazam Counts  \
0      18,004,655                 22,931          4,818,457      2,669,262   
1       7,780,028                 28,444          6,623,075      1,118,279   
2       5,022,621                  5,639          7,208,651      5,285,340   
3     190,260,277                203,384                NaN     11,822,942   
4       4,493,884                  7,006            207,179        457,017   

  TIDAL Popularity Explicit Track  
0              NaN              0  
1              NaN              1  
2              NaN              0  
3              NaN              0  
4              NaN              1  

[5 rows x 29 columns]

In [35]:
df.shape

(4600, 29)

## Descripción de las variables principales

### Información y tipos de datos

El dataset se compone de 14 columnas y 10000 entradas no nulas. Éste dataset ya se encuentra limpio y los tipos de datos corresponden a la información proporcionada en cada columna.

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4600 entries, 0 to 4599
Data columns (total 29 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Track                       4600 non-null   object 
 1   Album Name                  4600 non-null   object 
 2   Artist                      4595 non-null   object 
 3   Release Date                4600 non-null   object 
 4   ISRC                        4600 non-null   object 
 5   All Time Rank               4600 non-null   object 
 6   Track Score                 4600 non-null   float64
 7   Spotify Streams             4487 non-null   object 
 8   Spotify Playlist Count      4530 non-null   object 
 9   Spotify Playlist Reach      4528 non-null   object 
 10  Spotify Popularity          3796 non-null   float64
 11  YouTube Views               4292 non-null   object 
 12  YouTube Likes               4285 non-null   object 
 13  TikTok Posts                3427 

### Descripción estadistica del dataset

In [37]:
# Descripción de los datos numericos
df.describe()

Track Score  Spotify Popularity  Apple Music Playlist Count  \
count  4600.000000         3796.000000                  4039.00000   
mean     41.844043           63.501581                    54.60312   
std      38.543766           16.186438                    71.61227   
min      19.400000            1.000000                     1.00000   
25%      23.300000           61.000000                    10.00000   
50%      29.900000           67.000000                    28.00000   
75%      44.425000           73.000000                    70.00000   
max     725.400000           96.000000                   859.00000   

       Deezer Playlist Count  Amazon Playlist Count  TIDAL Popularity  \
count            3679.000000            3545.000000               0.0   
mean               32.310954              25.348942               NaN   
std                54.274538              25.989826               NaN   
min                 1.000000               1.000000               NaN   
25%                 5.000000               8.000000               NaN   
50%                15.000000              17.000000               NaN   
75%                37.000000              34.000000               NaN   
max               632.000000             210.000000               NaN   

       Explicit Track  
count     4600.000000  
mean         0.358913  
std          0.479734  
min          0.000000  
25%          0.000000  
50%          0.000000  
75%          1.000000  
max          1.000000

In [38]:
# Descripción de los datos no-numéricos
df.describe(exclude="number")

Track        Album Name Artist Release Date  \
count                   4600              4600   4595         4600   
unique                  4370              4005   1999         1562   
top     Danza Kuduro - Cover  Un Verano Sin Ti  Drake     1/1/2012   
freq                      13                20     63           38   

                ISRC All Time Rank Spotify Streams Spotify Playlist Count  \
count           4600          4600            4487                   4530   
unique          4598          4577            4425                   4207   
top     USWL11700269         3,441   1,655,575,417                      1   
freq               2             2               4                     46   

       Spotify Playlist Reach YouTube Views  ... TikTok Likes TikTok Views  \
count                    4528          4292  ...         3620         3619   
unique                   4478          4290  ...         3615         3616   
top                         3    30,913,276  ...       14,000  158,504,854   
freq                        8             2  ...            2            2   

       YouTube Playlist Reach AirPlay Spins SiriusXM Spins  \
count                    3591          4102           2477   
unique                   3458          3267            689   
top                   381,728             1              1   
freq                       13            69             54   

       Deezer Playlist Reach Pandora Streams Pandora Track Stations  \
count                   3672            3494                   3332   
unique                  3558            3491                   2975   
top                    1,097      56,972,562                      9   
freq                      17               2                      6   

       Soundcloud Streams Shazam Counts  
count                1267          4023  
unique               1265          4002  
top                    27             1  
freq                    2             5  

[4 rows x 22 columns]

### Observaciones sobre la información contenida en el dataset

Se pueden recoger las siguientes observaciones del dataset, en esta etapa preliminar.

- Existen 29 variables que se pueden relacionar con el desempeño de cada cancion en terminos de su "popularidad", siendo 23 numéricas y 6 categóricas
- El dataset requiere de limpieza ya que las columnas estan siendo identificadas con el tipo de dato incorrecto
